crates/astria-core/src/protocol/transaction/v1/action/mod.rs

```diff
@@ -1894,6 +1912,149 @@ enum BridgeSudoChangeErrorKind {
     InvalidFeeAsset(#[source] asset::ParseDenomError),
 }
 
+#[derive(Debug, Clone)]
+pub struct BridgeTransfer {
+    pub to: Address,
+    pub amount: u128,
+    // asset to use for fee payment.
+    pub fee_asset: asset::Denom,
+    // the address on the destination chain to send the transfer to.
+    pub destination_chain_address: String,
+    // the address of the bridge account to transfer from.
+    pub bridge_address: Address,
+    // The block number of the rollup block containing the withdrawal event.
+    pub rollup_block_number: u64,
+    // The identifier of the withdrawal event in the rollup block.
+    pub rollup_withdrawal_event_id: String,
+}
+
+impl Protobuf for BridgeTransfer {
+    type Error = BridgeTransferError;
+    type Raw = raw::BridgeTransfer;
+
+    #[must_use]
+    fn into_raw(self) -> raw::BridgeTransfer {
+        raw::BridgeTransfer {
+            to: Some(self.to.into_raw()),
+            amount: Some(self.amount.into()),
+            fee_asset: self.fee_asset.to_string(),
+            bridge_address: Some(self.bridge_address.into_raw()),
+            destination_chain_address: self.destination_chain_address,
+            rollup_block_number: self.rollup_block_number,
+            rollup_withdrawal_event_id: self.rollup_withdrawal_event_id,
+        }
+    }
+
+    #[must_use]
+    fn to_raw(&self) -> raw::BridgeTransfer {
+        raw::BridgeTransfer {
+            to: Some(self.to.to_raw()),
+            amount: Some(self.amount.into()),
+            fee_asset: self.fee_asset.to_string(),
+            bridge_address: Some(self.bridge_address.to_raw()),
+            destination_chain_address: self.destination_chain_address.clone(),
+            rollup_block_number: self.rollup_block_number,
+            rollup_withdrawal_event_id: self.rollup_withdrawal_event_id.clone(),
+        }
+    }
+
+    /// Convert from a raw, unchecked protobuf [`raw::BridgeTransferAction`].
+    ///
+    /// # Errors
+    ///
+    /// - if the `to` field is not set
+    /// - if the `to` field is invalid
+    /// - if the `amount` field is invalid
+    /// - if the `fee_asset` field is invalid
+    /// - if the `from` field is invalid
```

Comment:
should there also be validation that the `destination_chain_address` is set, as well as some restrictions on it's length?

Maybe this is more a sequencer concern, notice we don't have this on bridge lock but probably should? There's a limit to what a reasonable sized destination address is.

Reply:
added it to be unempty. we should enforce the nonempty/size constraint in the sequencer app, not just core, since users can use other tooling to create actions. and yes we should also have this on BridgeLock. i'll make an issue